OpenRA.Game/Traits/ActivityUtils.cs

```diff
@@ -44,7 +44,7 @@ public static Activity RunActivity(Actor self, Activity act)
 				else
 					start = current;
 
-				if (prev == act)
+				if (act == prev || act == prev.Parent)
```

Comment:
Not sure where the confusion is coming from. An activity's `Tick` method shouldn't explicitly return its `Parent`, the `NextActivity` getter already handles that. Also activities should generally never assume to know what their `Parent` is, even less so its grant parent So returning `Parent.Parent` would require the activity to know about implementation details that it doesn't have any business knowing.

The goal is to eventually have activities that are as small in scope and self-contained enough that they can be mixed and matched as needed.